src/api/user/userApiClient.ts

```diff
@@ -8,17 +8,24 @@ const userApi = axios.create({
   timeout: 2000,
 });
 
-// TODO: refreshToken을 담는 로직이 필요. 아직 API에 body인지 header인지 미정인 상태.
 // 사이클 방지 및 refresh를 사용할 곳이 여기뿐이라 이곳에서 정의.
-const refreshAccessToken = () => userApi.post<IRefreshResponse>('/refresh')
-  .then((res) => {
-    sessionStorage.setItem('accessToken', res.data.accessToken);
-    localStorage.setItem('refreshToken', res.data.refreshToken);
-  })
-  .catch(() => {
+const refreshAccessToken = async () => {
+  const refreshToken = localStorage.getItem('refreshToken');
+  if (!refreshToken) {
+    return Promise.reject();
+  }
+  return userApi.post<RefreshResponse>('/refresh', null, {
+    headers: {
+      refreshToken: `Bearer ${refreshToken}`,
+    },
+  }).then(({ data }) => {
+    sessionStorage.setItem('accessToken', data.accessToken);
+    localStorage.setItem('refreshToken', data.refreshToken);
+  }).catch(() => {
     sessionStorage.removeItem('accessToken');
     localStorage.removeItem('refreshToken');
   });
+};
```

Comment:
async 를 붙여줬는데 굳이 thenable 한 로직을 작성해야 하나 싶네요 🤔 

```suggestion
const refreshAccessToken = async () => {
  const refreshToken = localStorage.getItem("refreshToken");

  if (!refreshToken) {
    return Promise.reject();
  }

  try {
    const { data } = await userApi.post<RefreshResponse>("/refresh", null, {
      headers: { refreshToken: `Bearer ${refreshToken}` },
    });

    sessionStorage.setItem("accessToken", data.accessToken);
    localStorage.setItem("refreshToken", data.refreshToken);
  } catch () {
    sessionStorage.removeItem("accessToken");
    localStorage.removeItem("refreshToken");
  }
};
```

Reply:
막상 작성하니 then, catch의 가독성이 좋아보여서 저렇게 작성했네요 ㅋㅋ
프로젝트 전반적으로 async await문법을 주로 사용하니 맞추는게 좋겠네요.